arvo/eyre/tasks.md

```diff
@@ -0,0 +1,243 @@
+
+Many of the types referenced are detailed in the [Data Types](@/docs/arvo/eyre/data-types.md) document. It may also be useful to look at the `+clay` section of `/sys/lull.hoon` in Arvo where these `task`s, `gift`s and data structures are defined.
+
+The focus of this document is on interacting with Clay from userspace applications and threads, so it doesn't delve into the internal mechanics of Clay from a kernel development perspective.
+
+# `%live`
+
+This `task` notifies Eyre of the listening HTTPS and HTTP ports. It is automatically sent to Eyre by Vere and should not be used manually.
+
+## Accepts
+
+```hoon
+[insecure=@ud secure=(unit @ud)]
+```
+
+The `insecure` field is the HTTP port and `secure` is the optional HTTPS port.
+
+## Returns
+
+Eyre returns no `gift` in response to a `%live` `task`.
+
+# `%rule`
+
+This `task` either configures HTTPS with a certificate and keypair, or configures a DNS binding. You would typically do this via the `%acme` app rather than directly.
+
+## Accepts
+
+```hoon
+=http-rule
+```
+
+The [http-rule](@/docs/arvo/eyre/data-types.md#http-rule) is either tagged with `%cert` or `%turf`. A `%cert` `http-rule` sets an HTTPS certificate and keypair or removes it if null. A `%turf` `http-rule` either adds or removes a DNS binding depending on whether the `action` is `%put` or `%del`. Note that using `%turf` will automatically cause the system to try and obtain a certificate and keypair via Letsencrypt.
+
+## Returns
+
+Eyre returns no `gift` in response to a `%rule` `task`.
+
+# `%request`
+
+This `task` is how Eyre receives an inbound HTTP request. It will ordinarily be sent to Eyre by Vere so you wouldn't use it except perhaps in tests.
+
+## Accepts
+
+```hoon
+[secure=? =address =request:http]
+```
+
+The `secure` field says whether it's over HTTPS. The `address` is the ip address from which the request originated. The [request:http](@/docs/arvo/eyre/data-types.md#request-http) is the HTTP request itself containing the method, url, headers and body.
+
+## Returns
+
+Eyre may `pass` a `%response` `gift` on the appropriate `duct` depending on the contents of the `%request`, state of the connection and other factors.
+
+# `%request-local`
+
+This `task` is how Eyre receives an inbound HTTP request over the local loopback port. It behaves the same and takes the same arguments as in the [%request](#request) example except it skips any normally required authentication. Just like for a [%request](#request) `task`, you'd not normally use this manually.
+
+# `%cancel-request`
+
+This `task` is sent to Eyre by Vere when an HTTP client closes its previously established connection. You would not normally use this manually.
+
+## Accepts
+
+```hoon
+~
+```
+
+Takes no arguments.
+
+## Returns
```

Comment:
I agree, I initially wrote these along with the clay docs that way, but I changed it to bring it inline with the format of the existing docs. Probably should change all of them. @drbeefsupreme what do you think

Reply:
Yeah I agree, it does look better to just give the noun format given in `lull` directly rather than separating out the arguments. Especially when, someday, the code in the docs is pulled directly from the codebase. So I agree with these change, and we should go back and change all the other API docs (sorry for making you change the Clay docs!) to look like this.